### DISCUSSION TITLE
I created a service for my own needs based on BCU but worried about license compliance status

### DISCUSSION THREAD

**Discussion by zbalkan** [Klocman/Bulk-Crap-Uninstaller#563] [Q&A] [answered]
Hi,

This question is specific to @Klocman. I created a service for a minor problem: Win32_Product WMI class is a headache and for Group Policy WMI filters, it is unusable. The alternative is to use the path name of a known, static file, yet not every application is suitable for that. 

For instance, I want to apply a Group Policy to the computers if only Chrome is installed, I should write something like this:

```SQL
Select * from Win32_Product where Name like "Chrome"
```

But that is a huge headache. Some say [Win32_Product is evil](https://gregramsey.net/2012/02/20/win32_product-is-evil/) and MS actually says you should use something else like [this](https://techcommunity.microsoft.com/t5/ask-the-directory-services-team/how-to-not-use-win32-product-in-group-policy-filtering/ba-p/399983) or [this](https://techcommunity.microsoft.com/t5/ask-the-performance-team/group-policy-filtering-of-installed-applications/ba-p/374637).

Their suggestion is mostly using registry and write a MOF class based on them. And BCU shows us the registry is not always accurate. Other alternative is using a static path to a file -like full path to chrome.exe- to check. But not all paths are static. It is -almost- impossible to track user-scope installed MS Teams due to crazy path convention.

Then I decided to write my own WMI provider, a class with just enough properties. To publish it I started using Winget but at this point it is a hell to work with. Then I moved to BCUninstaller. I had to copy a part of the jungle to get a couple of bananas. But it works.

Now, I want to make it public though it needs refactoring. There are many unused methods, and possibly unused classes will appear after removing unreferenced methods. Before making it public, I want to get a review if the way I structured and documented are enough to comply with the Apache 2.0 license of BCU.

For a review I can make it publicly available for a while. I am not sure I can share while it is private.

What is next then?

**Comment by Klocman** [comment, marked as answer]
As long as you are going to publish the source and credit this repo or me then I don't see any issue with it.

**Reply by zbalkan** [reply]
Great. I'll clean up and make the repo public. Thanks for the heads up.

**Comment by zbalkan** [comment]
Hi Marcin,

I started some time ago but the development was halted. Here's the service I mentioned: https://github.com/zbalkan/AppLister
